subsys/bluetooth/host/Kconfig

```diff
@@ -101,7 +101,8 @@ config BT_RX_STACK_SIZE
 	depends on BT_HCI_HOST || BT_RECV_IS_RX_THREAD
 	default 512 if BT_HCI_RAW
 	default 2048 if BT_MESH
-	default 2048 if BT_SETTINGS
+	default 2048 if (BT_SETTINGS && !BT_SETTINGS_CCC_STORE_ON_WRITE)
+	default 2200 if BT_SETTINGS_CCC_STORE_ON_WRITE
```

Comment:
I don't really like the complex conditions for the default values of our Kconfig variables. That doesn't really scale if/when we start adding more features that either shrink or grow the needed stack. Should we just consider increasing the default from 2k to what you need? What exactly is it that's growing the stack consumption by so much? I don't see anything "big" in your patch, so if it's causing overflow then I'd say we were too close to the limit already to begin with, and so increasing the BT_SETTINGS default is justified unconditionally.

Reply:
I observe stack overflows happening after connected/disconnected/connected sequence. I didn't notice this without my patch, so assumed 2048 is OK for BT_SETTINGS only. But I also cannot see anything stack hungry in my changes, so we can go with 2200 for BT_SETTINGS.

Thanks for all suggestions, I will add style improvements after weekend ;)